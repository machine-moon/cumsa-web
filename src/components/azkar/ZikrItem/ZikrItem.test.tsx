import React from "react";
import { render } from "@testing-library/react";
import { ZikrItem } from "./index";

// Mock the AudioPlayer component
jest.mock("@/components/azkar/AudioPlayer", () => ({
  AudioPlayer: ({ id, isActive, loading }: { id: number; isActive: boolean; loading: boolean }) => (
    <div data-testid="audio-player" data-id={id} data-active={isActive} data-loading={loading}>
      {loading ? "Loading..." : isActive ? "Pause" : "Play"}
    </div>
  ),
}));

describe("ZikrItem", () => {
  const mockZikr = {
    ID: 1,
    ARABIC_TEXT: "سُبْحَانَ اللهِ وَبِحَمْدِهِ",
    LANGUAGE_ARABIC_TRANSLATED_TEXT: "Subhanallahi wa bihamdihi",
    TRANSLATED_TEXT: "Glory is to Allah and praise is to Him",
    REPEAT: 100,
    AUDIO: "https://example.com/zikr.mp3",
  };

  const defaultProps = {
    zikr: mockZikr,
    display: {
      arabic: true,
      transliteration: true,
      translation: true,
    },
    onPlayAudio: jest.fn(),
    isPlaying: jest.fn(() => false),
    audioLoading: false,
    audioError: null,
  };

  it("renders correctly with all display options", () => {
    const { container } = render(<ZikrItem {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with only Arabic", () => {
    const { container } = render(
      <ZikrItem
        {...defaultProps}
        display={{
          arabic: true,
          transliteration: false,
          translation: false,
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with only translation", () => {
    const { container } = render(
      <ZikrItem
        {...defaultProps}
        display={{
          arabic: false,
          transliteration: false,
          translation: true,
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly without audio", () => {
    const { container } = render(
      <ZikrItem
        {...defaultProps}
        zikr={{
          ...mockZikr,
          AUDIO: "",
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with single repeat", () => {
    const { container } = render(
      <ZikrItem
        {...defaultProps}
        zikr={{
          ...mockZikr,
          REPEAT: 1,
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with audio playing", () => {
    const { container } = render(
      <ZikrItem {...defaultProps} isPlaying={() => true} audioLoading={true} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with audio error", () => {
    const { container } = render(
      <ZikrItem {...defaultProps} audioError="Failed to load audio" isPlaying={() => true} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
