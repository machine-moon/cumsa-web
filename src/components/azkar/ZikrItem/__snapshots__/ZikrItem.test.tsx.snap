// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZikrItem renders correctly with all display options 1`] = `
<div
  class="relative p-4 rounded-lg bg-white/80 backdrop-blur border border-[var(--blue)]/25 flex flex-col gap-3 text-sm sm:text-base shadow-[0_2px_8px_-2px_rgba(0,0,0,0.08)] animate-fade-in"
>
  <div
    class="text-right font-medium leading-relaxed text-lg sm:text-xl"
    dir="rtl"
  >
    سُبْحَانَ اللهِ وَبِحَمْدِهِ
  </div>
  <div
    class="text-right leading-relaxed text-sm sm:text-base text-[var(--foreground)]/70"
    dir="rtl"
  >
    Subhanallahi wa bihamdihi
  </div>
  <div
    class="text-[var(--foreground)]/80 leading-relaxed text-xs sm:text-sm"
    dir="ltr"
  >
    Glory is to Allah and praise is to Him
  </div>
  <div
    class="flex items-center justify-between mt-1 gap-4"
  >
    <span
      class="text-xs text-[var(--muted)] min-h-[14px] flex items-center"
    >
      Repeat: 100 times
    </span>
    <div
      data-active="false"
      data-id="zikr-1"
      data-loading="false"
      data-testid="audio-player"
    >
      Play
    </div>
  </div>
</div>
`;

exports[`ZikrItem renders correctly with audio error 1`] = `
<div
  class="relative p-4 rounded-lg bg-white/80 backdrop-blur border border-[var(--blue)]/25 flex flex-col gap-3 text-sm sm:text-base shadow-[0_2px_8px_-2px_rgba(0,0,0,0.08)] animate-fade-in"
>
  <div
    class="text-right font-medium leading-relaxed text-lg sm:text-xl"
    dir="rtl"
  >
    سُبْحَانَ اللهِ وَبِحَمْدِهِ
  </div>
  <div
    class="text-right leading-relaxed text-sm sm:text-base text-[var(--foreground)]/70"
    dir="rtl"
  >
    Subhanallahi wa bihamdihi
  </div>
  <div
    class="text-[var(--foreground)]/80 leading-relaxed text-xs sm:text-sm"
    dir="ltr"
  >
    Glory is to Allah and praise is to Him
  </div>
  <div
    class="flex items-center justify-between mt-1 gap-4"
  >
    <span
      class="text-xs text-[var(--muted)] min-h-[14px] flex items-center"
    >
      Repeat: 100 times
    </span>
    <div
      data-active="true"
      data-id="zikr-1"
      data-loading="false"
      data-testid="audio-player"
    >
      Pause
    </div>
  </div>
</div>
`;

exports[`ZikrItem renders correctly with audio playing 1`] = `
<div
  class="relative p-4 rounded-lg bg-white/80 backdrop-blur border border-[var(--blue)]/25 flex flex-col gap-3 text-sm sm:text-base shadow-[0_2px_8px_-2px_rgba(0,0,0,0.08)] animate-fade-in"
>
  <div
    class="text-right font-medium leading-relaxed text-lg sm:text-xl"
    dir="rtl"
  >
    سُبْحَانَ اللهِ وَبِحَمْدِهِ
  </div>
  <div
    class="text-right leading-relaxed text-sm sm:text-base text-[var(--foreground)]/70"
    dir="rtl"
  >
    Subhanallahi wa bihamdihi
  </div>
  <div
    class="text-[var(--foreground)]/80 leading-relaxed text-xs sm:text-sm"
    dir="ltr"
  >
    Glory is to Allah and praise is to Him
  </div>
  <div
    class="flex items-center justify-between mt-1 gap-4"
  >
    <span
      class="text-xs text-[var(--muted)] min-h-[14px] flex items-center"
    >
      Repeat: 100 times
    </span>
    <div
      data-active="true"
      data-id="zikr-1"
      data-loading="true"
      data-testid="audio-player"
    >
      Loading...
    </div>
  </div>
</div>
`;

exports[`ZikrItem renders correctly with only Arabic 1`] = `
<div
  class="relative p-4 rounded-lg bg-white/80 backdrop-blur border border-[var(--blue)]/25 flex flex-col gap-3 text-sm sm:text-base shadow-[0_2px_8px_-2px_rgba(0,0,0,0.08)] animate-fade-in"
>
  <div
    class="text-right font-medium leading-relaxed text-lg sm:text-xl"
    dir="rtl"
  >
    سُبْحَانَ اللهِ وَبِحَمْدِهِ
  </div>
  <div
    class="flex items-center justify-between mt-1 gap-4"
  >
    <span
      class="text-xs text-[var(--muted)] min-h-[14px] flex items-center"
    >
      Repeat: 100 times
    </span>
    <div
      data-active="false"
      data-id="zikr-1"
      data-loading="false"
      data-testid="audio-player"
    >
      Play
    </div>
  </div>
</div>
`;

exports[`ZikrItem renders correctly with only translation 1`] = `
<div
  class="relative p-4 rounded-lg bg-white/80 backdrop-blur border border-[var(--blue)]/25 flex flex-col gap-3 text-sm sm:text-base shadow-[0_2px_8px_-2px_rgba(0,0,0,0.08)] animate-fade-in"
>
  <div
    class="text-[var(--foreground)]/80 leading-relaxed text-xs sm:text-sm"
    dir="ltr"
  >
    Glory is to Allah and praise is to Him
  </div>
  <div
    class="flex items-center justify-between mt-1 gap-4"
  >
    <span
      class="text-xs text-[var(--muted)] min-h-[14px] flex items-center"
    >
      Repeat: 100 times
    </span>
    <div
      data-active="false"
      data-id="zikr-1"
      data-loading="false"
      data-testid="audio-player"
    >
      Play
    </div>
  </div>
</div>
`;

exports[`ZikrItem renders correctly with single repeat 1`] = `
<div
  class="relative p-4 rounded-lg bg-white/80 backdrop-blur border border-[var(--blue)]/25 flex flex-col gap-3 text-sm sm:text-base shadow-[0_2px_8px_-2px_rgba(0,0,0,0.08)] animate-fade-in"
>
  <div
    class="text-right font-medium leading-relaxed text-lg sm:text-xl"
    dir="rtl"
  >
    سُبْحَانَ اللهِ وَبِحَمْدِهِ
  </div>
  <div
    class="text-right leading-relaxed text-sm sm:text-base text-[var(--foreground)]/70"
    dir="rtl"
  >
    Subhanallahi wa bihamdihi
  </div>
  <div
    class="text-[var(--foreground)]/80 leading-relaxed text-xs sm:text-sm"
    dir="ltr"
  >
    Glory is to Allah and praise is to Him
  </div>
  <div
    class="flex items-center justify-between mt-1 gap-4"
  >
    <span
      class="text-xs text-[var(--muted)] min-h-[14px] flex items-center"
    />
    <div
      data-active="false"
      data-id="zikr-1"
      data-loading="false"
      data-testid="audio-player"
    >
      Play
    </div>
  </div>
</div>
`;

exports[`ZikrItem renders correctly without audio 1`] = `
<div
  class="relative p-4 rounded-lg bg-white/80 backdrop-blur border border-[var(--blue)]/25 flex flex-col gap-3 text-sm sm:text-base shadow-[0_2px_8px_-2px_rgba(0,0,0,0.08)] animate-fade-in"
>
  <div
    class="text-right font-medium leading-relaxed text-lg sm:text-xl"
    dir="rtl"
  >
    سُبْحَانَ اللهِ وَبِحَمْدِهِ
  </div>
  <div
    class="text-right leading-relaxed text-sm sm:text-base text-[var(--foreground)]/70"
    dir="rtl"
  >
    Subhanallahi wa bihamdihi
  </div>
  <div
    class="text-[var(--foreground)]/80 leading-relaxed text-xs sm:text-sm"
    dir="ltr"
  >
    Glory is to Allah and praise is to Him
  </div>
  <div
    class="flex items-center justify-between mt-1 gap-4"
  >
    <span
      class="text-xs text-[var(--muted)] min-h-[14px] flex items-center"
    >
      Repeat: 100 times
    </span>
  </div>
</div>
`;
